http-netty/src/main/java/io/micronaut/http/netty/EventLoopSerializer.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2017-2024 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.http.netty;
+
+import io.micronaut.core.annotation.Internal;
+import io.netty.util.concurrent.EventExecutor;
+import io.netty.util.concurrent.OrderedEventExecutor;
+
+/**
+ * This class forwards reactive operations onto an {@link io.netty.channel.EventLoop} if they are
+ * called from outside that loop. It avoids unnecessary {@link EventExecutor#execute} calls when
+ * the reactive methods are called inside the loop. All the while it ensures the operations remain
+ * serialized (in the same order).
+ * <p>
+ * Should be used like this:
+ * <pre>
+ *     public void onNext(Object item) {
+ *         if (serializer.executeNow(() -> onNext0(item))) {
+ *             onNext0(item);
+ *         }
+ *     }
+ *
+ *     private void onNext0(Object item) {
+ *         ...
+ *     }
+ * </pre>
+ * <p>
+ * This class is <b>not</b> thread-safe: The invariants for calls to {@link #executeNow} are very
+ * strict. In particular:
+ * <ul>
+ *     <li>There must be no concurrent calls to {@link #executeNow}.</li>
+ *     <li>When {@link #executeNow} returns {@code true}, the subsequent execution of the child
+ *     method ({@code onNext0} in the above example) must fully complete before the next
+ *     {@link #executeNow} call. This ensures that there are no concurrent calls to the child
+ *     method.</li>
+ * </ul>
+ * Both of these invariants are guaranteed by the reactive spec, but may not apply to other use
+ * cases.
+ *
+ * @since 4.4.0
+ * @author Jonas Konrad
+ */
+@Internal
+public final class EventLoopSerializer {
```

Comment:
can we rename to `EventLoopFlow`, the name `Serializer` can be confused with serialization